screens/GameDetails.js

```diff
@@ -0,0 +1,301 @@
+import EStyleSheet from "react-native-extended-stylesheet";
+import { View, Text, TouchableOpacity, FlatList } from "react-native";
+import Button from "../common/Button";
+import BackArrow from "../common/BackArrow";
+import { DataStore } from "aws-amplify";
+import "@azure/core-asynciterator-polyfill";
+import {
+  Player,
+  Game,
+  Location,
+  GamePlayer,
+  Rsvp,
+  SkillLevel,
+} from "../src/models";
+
+import {epochToLocalDate} from '../utils/TimeUtil';
+import {epochToLocalTime} from '../utils/TimeUtil';
+import rsvp from "../utils/rsvp";
+
+function GameDetails({ route, navigation }) {
+  const details = route.params.item;
+  console.log("detuals", details)
+  const [loading, setLoading] = useState(false);
+  const [accepted, setAccepted] = useState([]);
+  const [declined, setDeclined] = useState([]);
+  const [thisPlayer, setThisPlayer] = useState([]);
+
+  const [gameOrganizer, setGameOrganizer] = useState([]);
+
+  //gets current user
+  async function getPlayer() {
+    setLoading(true);
+    try {
+      const response = await Auth.currentUserInfo();
+      const player = await DataStore.query(Player, (p) => p.email.eq(response.attributes.email));
+      console.log("This player returned: ", player[0]);
+      // setThisPlayer(player[0]);
+      return player[0];
+
+
+    } catch (error) {
+      console.log("Error getting player: ", error.message);
+    }
+  }
+
+  async function getInvitedPlayers() {
+    setLoading(true);
+    let playerids = details.invited_players;
+    console.log("player ids: ", playerids);
+
+    acceptedPlayers = [];
+    declinedPlayers = [];
+    for (let i = 0; i < playerids.length; i++) {
+      let playerId = playerids[i];
+      const gamePlayer = await DataStore.query(GamePlayer, (c) => c.and(c => [c.player_id.eq(playerId), c.game_id.eq(details.id)]));
+      console.log("game player returned: ", gamePlayer);
+      const player = await DataStore.query(Player, (c) => c.id.eq(playerId));
+      console.log("player returned: ", player);
+      if (gamePlayer[0].rsvp == Rsvp.ACCEPTED) {
+        acceptedPlayers.push({ id: player[0].id, name: player[0].name, status: "In" });
+      }
+      else {
+        declinedPlayers.push({ id: player[0].id, name: player[0].name, status: "Out" });
+      }
+    }
+
+    // setAccepted(acceptedPlayers);
+    // setDeclined(declinedPlayers);
+    // setLoading(false);
+    return {
+      accepted: acceptedPlayers,
+      declined: declinedPlayers
+    }
+  }
+
+  async function getGameOrganizer(player) {
+    if (player.id == details.organizer) {
+      return player.name;
+    } else {
+      const organizer = await DataStore.query(Player, details.organizer);
+      console.log("organizer returned: ", organizer);
+      return organizer.name;
+    }
+  }
+
+
+  useEffect(() => {
+    async function fetchData() {
+      await getPlayer();
+      await getGameOrganizer();
+      await getInvitedPlayers();
+    }
+  
+    fetchData();
+  }, []);
```

Comment:
Why do you have two useEffects that do the same thing? Get rid of this first one.